src/Commands/CleanCommand.php

```diff
@@ -126,20 +131,26 @@ protected function deleteOrphanedFiles()
             throw FileCannotBeAdded::diskDoesNotExist($diskName);
         }
 
-        $mediaIds = collect($this->mediaRepository->all()->pluck('id')->toArray());
-
-        collect($this->fileSystem->disk($diskName)->directories())
-            ->filter(function (string $directory) {
-                return is_numeric($directory);
-            })
-            ->reject(function (string $directory) use ($mediaIds) {
-                return $mediaIds->contains((int) $directory);
-            })->each(function (string $directory) use ($diskName) {
-                if (! $this->isDryRun) {
-                    $this->fileSystem->disk($diskName)->deleteDirectory($directory);
-                }
-
-                $this->info("Orphaned media directory `{$directory}` ".($this->isDryRun ? 'found' : 'has been removed'));
-            });
+        $medias = $this->mediaRepository->all();
+
+        $mediaPaths = $medias->map(function ($media) {
+            return rtrim($this->pathGenerator->getPath($media), '/');
+        });
+
+        $mediaPaths = $mediaPaths->merge($medias->map(function ($media) {
```

Comment:
Do we really need that `$mediaPaths` intermediate variable.

I've got nothing against a long collection chain.